scripts/cli/commands/generateScreen.ts

```diff
@@ -0,0 +1,190 @@
+  })
+
+  // @ts-expect-error: subValue not found on promptAnswer
+  const subValue = promptAnswer.subValue
+
+  // Return the result when user selects current directory
+  if (subValue === '.') {
+    return basePath
+  }
+  // Recursively execute path selection when subValue is not the current directory
+  const subResult = await selectPath(`${basePath}/${subValue}`)
+
+  return subResult
+}
+
+/**
+ * Validates if a route with the given name already exists in the specified path.
+ * @param routeName - The name of the route.
+ * @param routePath - The path where the route should be generated.
+ * @throws Error if the route already exists in the specified path.
+ */
+const validateRoute = (routeName: string, routePath: string) => {
+  const filePath = `${routePath}/${routeName}.tsx`
+  if (fs.existsSync(filePath)) {
+    logger.error(`Route ${routeName} already exists in ${routePath}`)
+  }
+}
+
+/**
+ * Creates a route file with the given route name and path.
+ * @param {string} routeName - The name of the route.
+ * @param {string} routePath - The path where the route file will be created.
+ */
+const createRouteFile = (routeName: string, routePath: string) => {
+  const screenName = `${routeName.charAt(0).toUpperCase() + routeName.slice(1)}Screen`
+  fs.writeFileSync(
+    `${routePath}/${routeName.toLowerCase()}.tsx`,
+    `import { ${screenName} } from '@baca/screens'
+
+export default ${screenName}
+`
+  )
+}
+
+/**
+ * Validates if a screen with the given name already exists.
+ * @param screenName - The name of the screen.
+ * @throws Error if the screen already exists.
+ */
+const validateScreen = (screenName: string) => {
+  const filePath = `${SCREENS_DIRECTORY}/${screenName}.tsx`
+  if (fs.existsSync(filePath)) {
+    logger.error(`Screen ${screenName} already exists`)
+  }
+}
+
+/**
+ * Creates a screen file with the given screen name.
+ * @param {string} screenName - The name of the screen.
+ */
+const createScreenFile = (screenName: string) => {
+  const screenFromFile = fs.readFileSync('./templates/screen_template.tsx', 'utf8')
+  const screenContent = screenFromFile.replaceAll('_NAME_', screenName)
+
+  fs.writeFileSync(`${SCREENS_DIRECTORY}/${screenName}.tsx`, screenContent)
+}
+
+const addToScreensIndex = (screenName: string) => {
+  const indexFilePath = `${SCREENS_DIRECTORY}/index.ts`
+  const indexFile = fs.readFileSync(indexFilePath, 'utf8')
+  const newIndexFile = indexFile.padEnd(indexFile.length) + `export * from './${screenName}'\n`
+
+  fs.writeFileSync(indexFilePath, newIndexFile)
+}
+
+const promptTabName = async () => {
+  const promptAnswer = await prompt({
+    message: 'What is your screen name?',
+    name: 'tabName',
+    type: 'input',
+  })
+
+  // @ts-expect-error: generator not found on promptAnswer
+  const tabName = promptAnswer.tabName as string
+
+  if (!tabName) {
+    throw new Error('Tab name is required')
+  }
+  return tabName
+}
+
+const createNewNavTab = (tabName: string) => {
```

Comment:
Something is not working here 🤔 
<img width="820" alt="Screenshot 2024-04-22 at 08 45 13" src="https://github.com/binarapps/baca-react-native-template/assets/60507270/971d8a95-4e43-441e-a682-7428a76cf593">